### DISCUSSION TITLE
How can you control the charge of a battery or change the PV output

### DISCUSSION THREAD

**Discussion by ripcam21** [tijsverkoyen/HomeAssistant-FusionSolar#121] [General]
Not an issue, rather a comment on extending this integration to allow communication to the inverter to allow the home assistant to charge and discharge battery at the owners discretion.  Also having the ability to up and lower the solar PV output when feed-in prices are high and low.

Can this be done?

BTW, great work on the integration.

**Comment by tijsverkoyen** [comment]
At this point the integration only reads data. 
I also don't know if this is possible. I don't own a battery, so I don't have the system to be able to test while developing it.

Feel free to create a PR if you are able to.

**Comment by ripcam21** [comment]
Unfortunately my skills are lacking here to add any meaningful development... hopefully you purchase a battery which might inspire further development :).  maybe someone else who has the skills can look into this.

**Comment by tijsverkoyen** [comment]
Probably when my digital meter is placed. At the moment the grid is my infinite battery ;-)

But to be honest: if you want to do things with HA and batteries, to optimize costs. I think this integration is not "fast" enough. The data can get "slow" if you have multiple device groups. See the expert from README.md

> The realtime data is updated every minute per device group. As the API only allows 1 call per minute to each endpoint and the same endpoint is needed for each device group. The more different devices you have the slower the updates will be. See Disabling devices

An option would be to use the modbus integration, which will give you actual realtime data and will be fully locally. See: https://github.com/wlcrs/huawei_solar. And this integration also allows you to configure stuff on the inverter.

**Comment by ripcam21** [comment]
Thanks for this.  I just read through the documentation and it looks like an option.  I'll try it out and see how I go.  Thanks again.

**Comment by nrasd** [comment]
Any news on this? Looking for a way to connect to several plants once every day via internet and change parameters.
